### PR TITLE
fix: next.js 15 compatibility

### PR DESCRIPTION
**Pull request by thucpn**
Nextjs 15 moved `serverComponentsExternalPackages` from experimental to stable.
Also renamed from `serverComponentsExternalPackages` to `serverExternalPackages`

https://nextjs.org/docs/app/api-reference/next-config-js/serverExternalPackages

### CODE REVIEW

File: .changeset/good-tips-obey.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"llamaindex": patch
```
Comment (marcusschiesser) [inline .changeset/good-tips-obey.md:2]: @himself65 we saw this issue when using nextjs 15 as a warning in create-llama generated code. you might want to release this with the next minor version
```suggestion
"llamaindex": minor
```

Comment (himself65) [reply .changeset/good-tips-obey.md:2]: could we both suport nextjs 14 and 15?

Comment (himself65) [reply .changeset/good-tips-obey.md:2]: I don't want to bump to 0.9.0

Comment (marcusschiesser) [reply .changeset/good-tips-obey.md:2]: @himself65 nextjs config object and webpack config object both don't contain nextjs version, so we better wait to merge this PR once nextjs 15 got more popular - right now it works for nextjs 14 and 15 and for 15, we're just getting a warning.

**Pull request merged**